api/printful/printful.php

```diff
@@ -8,12 +8,24 @@ public function __construct($printful_api_key = PRINTFUL_API_KEY) {
     }
 
     protected function send_request($URI, $data = false) {
+        $endpoint = 'https://api.printful.com/' . $URI;
+        $isOrder =substr($URI, 0, 6) == 'orders';
+        if(!$isOrder){ // check cache, ignore orders because they shouldn't be cached
+            $r = getCached($endpoint);
+            if($r!== null){ // endpoint is in cache

```

Comment:
I'd like for the api-cache to be able to handle the cache checking, returning and insertion without needing to implement this everywhere we use the cache. Could the `getCached` method take a instance of `$ch` here where the client just sets the curlopts and it sends the request if needed?

Reply:
Changed in 7cb4d48. Only caveat is that I made a new object to encapsulate some cURL info and return the object instead of just the content. I only did that because the `PrintfulCustomAPIException`s checked the HTTP response code, which isn't 200 if a cached item is returned (because `curl_exec` wasn't called).
I can have it pass the result only if we switch to throwing generic exceptions instead. Or we could pass the exception type name as a string in the parameter? Let me know if you want one of these instead of the current implementation